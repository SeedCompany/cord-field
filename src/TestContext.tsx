import { ApolloProvider } from '@apollo/client';
import {
  Request as ExpressRequest,
  Response as ExpressResponse,
} from 'express';
import { Request } from 'jest-express/lib/request';
import { Response } from 'jest-express/lib/response';
import { useMemo, useState } from 'react';
import { HelmetProvider } from 'react-helmet-async';
import { StaticRouter } from 'react-router-dom/server';
import { ChildrenProp } from '~/common';
import { createClient } from './api/client/createClient';
import { Nest } from './components/Nest';
import { RequestContext } from './hooks';

export const TestContext = ({
  url,
  children,
}: { url: string } & ChildrenProp) => {
  // @ts-expect-error yes the type doesn't match we are faking it.
  const req: ExpressRequest = useMemo(() => new Request(url), [url]);
  const res = new Response() as unknown as ExpressResponse;
  const [client] = useState(() => createClient({ ssr: { req, res } }));
  return (
    <Nest
      elements={[
        <HelmetProvider key="helmet" context={{}} children={[]} />,
        <RequestContext.Provider key="req" value={req} />,
        <StaticRouter key="router" location={req.originalUrl} />,
        <ApolloProvider key="apollo" client={client} children={[]} />,
      ]}
      children={children}
    />
  );
};
