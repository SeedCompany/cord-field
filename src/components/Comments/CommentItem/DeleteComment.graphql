mutation DeleteComment($id: ID!) {
  deleteComment(id: $id) {
    __typename
  }
}
