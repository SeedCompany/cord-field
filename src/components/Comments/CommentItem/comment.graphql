fragment comment on Comment {
  id
  body {
    value
    canEdit
    canRead
  }
  creator {
    id
    fullName
    avatarLetters
  }
  createdAt
  modifiedAt
  canDelete
}
