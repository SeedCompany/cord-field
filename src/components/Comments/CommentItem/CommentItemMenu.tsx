import { Delete as DeleteIcon, Edit as EditIcon } from '@mui/icons-material';
import {
  ListItemIcon,
  ListItemText,
  Menu,
  MenuItem,
  MenuProps,
} from '@mui/material';
import { CommentFragment } from './comment.graphql';

interface CommentItemMenuProps extends MenuProps {
  threadId?: string;
  comment: CommentFragment;
  onDelete: () => void;
  onEdit: () => void;
}

export const CommentItemMenu = ({
  comment,
  threadId,
  onDelete,
  onEdit,
  ...rest
}: CommentItemMenuProps) => {
  const listItemSx = {
    marginRight: 2,
    minWidth: 'unset',
  };

  if (!comment.canDelete && !comment.body.canEdit) {
    return null;
  }

  return (
    <Menu
      id={`${comment.id}-options-menu`}
      anchorOrigin={{ vertical: 'top', horizontal: 'right' }}
      transformOrigin={{ vertical: 'top', horizontal: 'right' }}
      {...rest}
    >
      {comment.body.canEdit && (
        <MenuItem onClick={onEdit}>
          <ListItemIcon sx={listItemSx}>
            <EditIcon fontSize="small" />
          </ListItemIcon>
          <ListItemText primary="Edit" />
        </MenuItem>
      )}
      {comment.canDelete && (
        <MenuItem onClick={onDelete}>
          <ListItemIcon sx={listItemSx}>
            <DeleteIcon fontSize="small" />
          </ListItemIcon>
          <ListItemText primary="Delete" />
        </MenuItem>
      )}
    </Menu>
  );
};
