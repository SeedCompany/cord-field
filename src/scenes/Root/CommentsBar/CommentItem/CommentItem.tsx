import { useMutation } from '@apollo/client';
import { MoreVert } from '@mui/icons-material';
import { Box, IconButton, MenuProps, Paper, Typography } from '@mui/material';
import Blocks from 'editorjs-blocks-react-renderer';
import { useState } from 'react';
import { removeItemFromList } from '~/api';
import { useDateTimeFormatter } from '~/components/Formatters';
import { CommentItemMenu } from '../CommentItemMenu';
import {
  CommentPropsFragment,
  CommentThreadPropsFragment,
  DeleteCommentDocument,
} from '../CommentsBar.graphql';

export interface CommentProps {
  comment: CommentPropsFragment;
  repliesCount?: number;
  isChild?: boolean;
  resourceId: string;
  isExpanded?: boolean;
  parent: CommentThreadPropsFragment;
  handleDeleteComment?: (comment: CommentPropsFragment) => Promise<void> | void;
  handleEditComment?: (comment: CommentPropsFragment) => Promise<void> | void;
  handleExpand?: (threadId: string | null) => void;
}

export const CommentItem = ({
  comment,
  repliesCount = 0,
  isChild,
  handleDeleteComment,
  isExpanded,
  handleExpand,
  parent,
}: CommentProps) => {
  const [actionsAnchor, setActionsAnchor] = useState<MenuProps['anchorEl']>();
  const [deleteCommentMutation] = useMutation(DeleteCommentDocument, {});

  const dateTimeFormatter = useDateTimeFormatter();
  const createdAtString = dateTimeFormatter(comment.createdAt);

  const deleteComment = async (commentId: string) => {
    const { data } = await deleteCommentMutation({
      variables: {
        id: commentId,
      },
      update: removeItemFromList({
        listId: [parent, 'comments'],
        item: comment,
      }),
    });

    if (data?.deleteComment) {
      setActionsAnchor(null);
      void handleDeleteComment?.(comment);
    }
  };

  return (
    <Paper
      elevation={repliesCount ? 4 : isChild ? 1 : 3}
      sx={[
        { padding: 2, width: '100%', position: 'relative' },
        { zIndex: 11 },
        !!isChild && { pl: 6, zIndex: 10 },
      ]}
      square
    >
      <Box
        sx={{
          display: 'flex',
          alignItems: 'center',
          mb: 1,
        }}
      >
        <Typography variant="body1" color="text.primary">
          {comment.creator.fullName}
        </Typography>
        <Typography
          variant="body2"
          color={(theme) => theme.palette.grey[500]}
          sx={{
            ml: 1,
          }}
        >
          {createdAtString}
        </Typography>
      </Box>
      <Box
        sx={{
          '& p': {
            fontSize: '0.875rem',
          },
          '& h1, & h2, & h3, & h4, & h5, & h6': {
            marginTop: '0',
          },
          '& .cdx-marker': {
            backgroundColor: 'rgba(245,235,111,0.5)',
            padding: '0',
          },
        }}
      >
        {comment.body.value && <Blocks data={comment.body.value} />}

        {repliesCount > 0 && (
          <Typography
            variant="body2"
            color="text.secondary"
            component="span"
            sx={{
              '&:hover': {
                cursor: 'pointer',
                textDecoration: 'underline',
              },
            }}
            onClick={() => {
              const newValue = (!isExpanded ? parent.id : null) || null;

              handleExpand?.(newValue);
            }}
          >
            {isExpanded ? 'Hide' : 'View'} Replies ({repliesCount})
          </Typography>
        )}
      </Box>
      <CommentItemMenu
        commentId={comment.id}
        anchorEl={actionsAnchor}
        onClose={() => setActionsAnchor(null)}
        open={Boolean(actionsAnchor)}
        onDelete={() => {
          void deleteComment(comment.id);
        }}
        onEdit={() => {
          setActionsAnchor(null);
        }}
      />

      <IconButton
        onClick={(e) => setActionsAnchor(e.currentTarget)}
        sx={(theme) => ({
          margin: theme.spacing(1),
          position: 'absolute',
          right: 0,
          top: 0,
        })}
      >
        <MoreVert
          sx={{
            fontSize: '1.25rem',
          }}
        />
      </IconButton>
    </Paper>
  );
};
