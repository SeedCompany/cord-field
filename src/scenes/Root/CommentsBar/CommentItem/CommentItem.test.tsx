import { queryByTestId, render } from '@testing-library/react';
import { DateTime } from 'luxon';
import { TestContext } from '~/TestContext';
import {
  CommentPropsFragment,
  CommentThreadPropsFragment,
} from '../CommentsBar.graphql';
import { CommentItem } from './CommentItem';

const commentText = 'This is a comment';

const comment: CommentPropsFragment = {
  __typename: 'Comment',
  id: '1',
  createdAt: DateTime.fromISO('2021-01-01T00:00:00.000Z'),
  modifiedAt: DateTime.fromISO('2021-01-01T00:00:00.000Z'),
  canDelete: true,
  creator: {
    id: '1',
    fullName: 'Full Name',
    avatarLetters: 'FN',
  },
  body: {
    value: {
      time: 1666023476823,
      blocks: [
        {
          id: 'nDNN-GePQ1',
          type: 'paragraph',
          data: {
            text: commentText,
          },
        },
      ],
      version: '2.24.3',
    },
  },
};

const parent: CommentThreadPropsFragment = {
  __typename: 'CommentThread',
  id: '1',
  firstComment: comment,
  comments: {
    __typename: 'CommentList',
    items: [comment],
    total: 1,
  },
};

const resourceId = '1';

test('matches snapshot', () => {
  const tree = render(
    <TestContext url={`/projects/${resourceId}`}>
      <CommentItem comment={comment} resourceId={resourceId} parent={parent} />
    </TestContext>
  );

  expect(tree).toMatchSnapshot();
  const body = tree.getByTestId('comment-body');
  expect(body).toBeInTheDocument();
  expect(body).toHaveTextContent(commentText);
});

test('menu options do not exist when can not delete', () => {
  const commentNoOptions: CommentPropsFragment = {
    ...comment,
    canDelete: false,
  };

  const parentNoOptions: CommentThreadPropsFragment = {
    ...parent,
    firstComment: commentNoOptions,
    comments: {
      __typename: 'CommentList',
      items: [commentNoOptions],
      total: 1,
    },
  };

  const tree = render(
    <TestContext url={`/projects/${resourceId}`}>
      <CommentItem
        comment={commentNoOptions}
        resourceId={resourceId}
        parent={parentNoOptions}
      />
    </TestContext>
  );

  expect(queryByTestId(tree.container, 'comment-menu-button')).toBeNull();
});

test('menu options exist when can delete', () => {
  const tree = render(
    <TestContext url={`/projects/${resourceId}`}>
      <CommentItem comment={comment} resourceId={resourceId} parent={parent} />
    </TestContext>
  );

  expect(queryByTestId(tree.container, 'comment-menu-button')).not.toBeNull();
});
