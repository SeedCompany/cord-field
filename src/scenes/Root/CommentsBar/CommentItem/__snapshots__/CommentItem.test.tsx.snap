// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  box-shadow: 0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12);
  padding: 16px;
  width: 100%;
  position: relative;
  z-index: 11;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 8px;
}

.emotion-2 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: rgba(0, 0, 0, 0.87);
}

.emotion-3 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  color: #9e9e9e;
  margin-left: 8px;
}

.emotion-4 p {
  font-size: 0.875rem;
}

.emotion-4 h1,
.emotion-4 h2,
.emotion-4 h3,
.emotion-4 h4,
.emotion-4 h5,
.emotion-4 h6 {
  margin-top: 0;
}

.emotion-4 .cdx-marker {
  background-color: rgba(245,235,111,0.5);
  padding: 0;
}

.emotion-4>* {
  padding: 0;
  margin: 0;
}

.emotion-5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  margin: 8px;
  position: absolute;
  right: 0;
  top: 0;
}

.emotion-5::-moz-focus-inner {
  border-style: none;
}

.emotion-5.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-5 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-5:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-5:hover {
    background-color: transparent;
  }
}

.emotion-5.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-6 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  font-size: 1.25rem;
}

.emotion-7 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-8 {
  position: fixed;
  z-index: 1300;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  visibility: hidden;
}

.emotion-9 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  background-color: transparent;
  -webkit-tap-highlight-color: transparent;
  z-index: -1;
}

.emotion-10 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  max-height: calc(100% - 96px);
  -webkit-overflow-scrolling: touch;
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
  position: absolute;
  overflow-y: auto;
  overflow-x: hidden;
  min-width: 16px;
  min-height: 16px;
  max-width: calc(100% - 32px);
  max-height: calc(100% - 32px);
  outline: 0;
}

.emotion-11 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
  outline: 0;
}

.emotion-12 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-height: 48px;
  padding-top: 6px;
  padding-bottom: 6px;
  box-sizing: border-box;
  white-space: nowrap;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-12::-moz-focus-inner {
  border-style: none;
}

.emotion-12.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-12 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-12:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-12:hover {
    background-color: transparent;
  }
}

.emotion-12.Mui-selected {
  background-color: rgba(25, 118, 210, 0.08);
}

.emotion-12.Mui-selected.Mui-focusVisible {
  background-color: rgba(25, 118, 210, 0.2);
}

.emotion-12.Mui-selected:hover {
  background-color: rgba(25, 118, 210, 0.12);
}

@media (hover: none) {
  .emotion-12.Mui-selected:hover {
    background-color: rgba(25, 118, 210, 0.08);
  }
}

.emotion-12.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-12.Mui-disabled {
  opacity: 0.38;
}

.emotion-12+.MuiDivider-root {
  margin-top: 8px;
  margin-bottom: 8px;
}

.emotion-12+.MuiDivider-inset {
  margin-left: 52px;
}

.emotion-12 .MuiListItemText-root {
  margin-top: 0;
  margin-bottom: 0;
}

.emotion-12 .MuiListItemText-inset {
  padding-left: 36px;
}

.emotion-12 .MuiListItemIcon-root {
  min-width: 36px;
}

@media (min-width:600px) {
  .emotion-12 {
    min-height: auto;
  }
}

.emotion-13 {
  min-width: 56px;
  color: rgba(0, 0, 0, 0.54);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-right: 16px;
  min-width: unset;
}

.emotion-14 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.25rem;
}

.emotion-15 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-width: 0;
  margin-top: 4px;
  margin-bottom: 4px;
}

.emotion-16 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  display: block;
}

<body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation3 emotion-0"
      >
        <div
          class="MuiBox-root emotion-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-2"
          >
            Full Name
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2 emotion-3"
          >
            1/1/2021, 5:00 AM
          </p>
        </div>
        <div
          class="MuiBox-root emotion-4"
        >
          <p>
            This is a comment
          </p>
        </div>
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-5"
          data-testid="comment-menu-button"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-6"
            data-testid="MoreVertIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root emotion-7"
          />
        </button>
      </div>
    </div>
    <div
      aria-hidden="true"
      class="MuiPopover-root MuiMenu-root MuiModal-root MuiModal-hidden emotion-8"
      id="1-options-menu"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiBackdrop-invisible emotion-9"
        style="opacity: 0; visibility: hidden;"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiPaper-root MuiMenu-paper MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiPopover-paper emotion-10"
        style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
        tabindex="-1"
      >
        <ul
          class="MuiList-root MuiList-padding MuiMenu-list emotion-11"
          role="menu"
          tabindex="-1"
        >
          <li
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters emotion-12"
            role="menuitem"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root emotion-13"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall emotion-14"
                data-testid="EditIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root emotion-15"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-16"
              >
                Edit
              </span>
            </div>
            <span
              class="MuiTouchRipple-root emotion-7"
            />
          </li>
          <li
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters emotion-12"
            role="menuitem"
            tabindex="-1"
          >
            <div
              class="MuiListItemIcon-root emotion-13"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall emotion-14"
                data-testid="DeleteIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root emotion-15"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-16"
              >
                Delete
              </span>
            </div>
            <span
              class="MuiTouchRipple-root emotion-7"
            />
          </li>
        </ul>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": .emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  box-shadow: 0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12);
  padding: 16px;
  width: 100%;
  position: relative;
  z-index: 11;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 8px;
}

.emotion-2 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: rgba(0, 0, 0, 0.87);
}

.emotion-3 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  color: #9e9e9e;
  margin-left: 8px;
}

.emotion-4 p {
  font-size: 0.875rem;
}

.emotion-4 h1,
.emotion-4 h2,
.emotion-4 h3,
.emotion-4 h4,
.emotion-4 h5,
.emotion-4 h6 {
  margin-top: 0;
}

.emotion-4 .cdx-marker {
  background-color: rgba(245,235,111,0.5);
  padding: 0;
}

.emotion-4>* {
  padding: 0;
  margin: 0;
}

.emotion-5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  margin: 8px;
  position: absolute;
  right: 0;
  top: 0;
}

.emotion-5::-moz-focus-inner {
  border-style: none;
}

.emotion-5.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-5 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-5:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-5:hover {
    background-color: transparent;
  }
}

.emotion-5.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-6 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  font-size: 1.25rem;
}

.emotion-7 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation3 emotion-0"
    >
      <div
        class="MuiBox-root emotion-1"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-2"
        >
          Full Name
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 emotion-3"
        >
          1/1/2021, 5:00 AM
        </p>
      </div>
      <div
        class="MuiBox-root emotion-4"
      >
        <p>
          This is a comment
        </p>
      </div>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-5"
        data-testid="comment-menu-button"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-6"
          data-testid="MoreVertIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root emotion-7"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
